control_panel_api/authentication.py

```diff
@@ -1,37 +1,69 @@
+import logging
+
+import requests
 from django.conf import settings
-import jwt
-from jwt.exceptions import InvalidTokenError
+from jose import jwt
+from jose.exceptions import JWTError
+from requests.exceptions import RequestException
 from rest_framework.authentication import (
     BaseAuthentication,
-    get_authorization_header
+    get_authorization_header,
 )
 from rest_framework.exceptions import AuthenticationFailed
 
 from control_panel_api.models import User
 
+logger = logging.getLogger(__name__)
 
-class Auth0JWTAuthentication(BaseAuthentication):
 
-    def authenticate(self, request):
+def get_jwks():
+    response = requests.get(settings.OIDC_WELL_KNOWN_URL)
+    return response.json()
+
+
+def get_key(jwks, unverified_header):
+    for jwk in jwks.get('keys'):
+        if jwk['kid'] == unverified_header['kid']:
+            return jwk
 
+    return None
```

Comment:
You may just raise `AuthenticationFailed` from here so you don't have to check the return value of this method outside (and save few lines of code).

Reply:
I wouldn't do that - it's mixing responsibilities. Getting a matching key out of a list doesn't technically have anything to do with authentication.

Reply:
Yes as mentioned in other comment, as these functions are outside the `Auth0JWTAuthentication` class raise an `AuthenticationFailed` here is a bit odd.